Holovibes/cuda_includes/reduce.cuh

```diff
@@ -1,25 +1,259 @@
-/*! \file
+/*! \file reduce.cuh
  *
- * \brief #TODO Add a description for this file
+ *  \brief Generic functions to apply optimized reduction operations on GPU side. Particuarly useful for min, max,
+ *  add,.. or any of these other kind of operations.
  */
 #pragma once
 using uint = unsigned int;
 
-/*! \brief Reduce add operation
+#include "cuda.h"
+#include "cuda_runtime.h"
+
+#include <cassert>
+#include <limits>
+
+#include "common.cuh"
+#include "cuda_memory.cuh"
```

Comment:
Shouldn't the includes be at the beginning of the file ? (after the pragma once)